api/src/main/java/ai/djl/util/Utils.java

```diff
@@ -351,4 +354,36 @@ public static Path getCacheDir() {
         }
         return Paths.get(cacheDir);
     }
+
+    /**
+     * Returns a string representation of the passed {@link Block} describing the input axes, output
+     * axes, and the block's children.
+     *
+     * @param block the block to describe
+     * @param blockName the name to be used for the passed block, or <code>null</code> if its class
+     *     name is to be used
+     * @param beginAxis skips all axes before this axis; use <code>0</code> to print all axes and
+     *     <code>1</code> to skip the batch axis.
+     * @return the string representation
+     */
+    public static String desribe(Block block, String blockName, int beginAxis) {
```

Comment:
I like a lot of your changes here. Instead of creating a second Block display function with `Block.toString()`, you should combine the two and move this definition directly into Block. Because this one has arguments, probably just implement `Block.toString() = describe(this, null, 0)` so both versions are unified

Reply:
Makes sense, we should just handle the uninitialized case better in `describe` so it can be used in `Block.toString` at any time, I will fix it.

Reply:
We should still have `describe ` as a static utility method in `Utils`, because my purpose here was also that other devs can just copy&pase `describe ` as template and then modify textual output to a form they better can read, i.e. the curly braces may not be everybody's style. I thought about introducing a template mechanism here, but this gets quickly complicated and copy&paste of `describe ` is probably the easiest for other people to achieve their desired formatting.

Reply:
If that is what you are thinking, I wonder if we should put this into a separate Utils file. Then, we can add a comment either in the Class or here indicating that users may want to try copying it. Otherwise, I don't know how many users would think of trying to build their own block display formatter

Reply:
I think `Utils `is fine for now. I am also thinking about the example to convert a DJL model into some connected flow graph for some javascript library for a better visual representation, maybe even 3 dimensional. So this `describe ` exposes a very simple visitor pattern.